beacon-chain/powchain/log_processing.go

```diff
@@ -301,19 +307,40 @@ func (s *Service) requestBatchedLogs(ctx context.Context) error {
 // requestMissingLogs requests any logs that were missed by requesting from previous blocks
 // until the current block(exclusive).
 func (s *Service) requestMissingLogs(ctx context.Context, blkNumber uint64, wantedIndex int64) error {
+	// prevent this method from being called recursively
+	s.requestingOldLogs = true
+	defer func() {
+		s.requestingOldLogs = false
+	}()
 	// We request from the last requested block till the current block(exclusive)
 	beforeCurrentBlk := big.NewInt(int64(blkNumber) - 1)
-
-	for i := s.lastRequestedBlock.Uint64() + 1; i <= beforeCurrentBlk.Uint64(); i++ {
-		err := s.ProcessETH1Block(ctx, big.NewInt(int64(i)))
+	startBlock := s.lastRequestedBlock.Uint64() + 1
+	for {
+		err := s.processBlksInRange(ctx, startBlock, beforeCurrentBlk.Uint64())
 		if err != nil {
 			return err
 		}
+
+		if s.lastReceivedMerkleIndex == wantedIndex {
+			break
+		}
+
+		// if the required logs still do not exist after the lookback period, then we return an error.
+		if startBlock < s.lastRequestedBlock.Uint64()-eth1LookBackPeriod {
+			return fmt.Errorf("despite requesting missing logs, latest index observed is not accurate. "+
```

Comment:
the multi-line error is kinda weird, maybe we can shorten it and put it in one line? Also let's avoid periods